core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2732,19 +2734,31 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         _.filterNot(_.value.startsWith("resourcemanager."))
       )
 
-      bucketIamResults <- gcsDAO.testSAGoogleBucketIam(
-        GcsBucketName(workspace.bucketName),
-        petKey,
-        expectedGoogleBucketPermissions
-      )
+      bucketIamResults <- gcsDAO
+        .testSAGoogleBucketIam(
+          GcsBucketName(workspace.bucketName),
+          petKey,
+          expectedGoogleBucketPermissions
+        )
+        .recoverWith { case t: StorageException =>
+          // Throw with the status code of the exception (for example 403 for invalid billing, 400 for requester pays)
+          // instead of a 500 to avoid Sentry notifications.
+          Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(t.getCode, t)))
```

Comment:
Should this be t.getMessage rather than the exception itself?

Reply:
You get more information if you pass the throwable itself. Do you have concerns about that information being included?

```
  def apply(statusCode: StatusCode, throwable: Throwable)(implicit source: ErrorReportSource): ErrorReport =
    ErrorReport(source.source,
                message(throwable),
                Some(statusCode),
                causes(throwable),
                throwable.getStackTrace,
                Option(throwable.getClass)
    )

Reply:
Meh, ok. I have misgivings about our continual surfacing of stack traces in the UI, but that's a different discussion.